src/stk/common/stack_trace.h

```diff
@@ -2,32 +2,91 @@
 
 #include <sstream>
 
-#if defined(__GNUC__) && defined(__linux__)
+#if defined(__linux__) && defined(STK_USE_BACKWARD)
 
-// Relying on GNU extensions
+#include <backward.hpp>
+
+void get_stack_trace(std::ostringstream& s, const int skip = 0)
+{
+    s << std::endl;
+    backward::StackTrace st;
+    st.load_here(128);
+    st.skip_n_firsts(1 + skip); // skip current frame
+    backward::Printer p;
+    p.object = p.address = true;
+    p.color_mode = backward::ColorMode::always;
+    p.print(st, s);
+}
+
+#elif  defined(__GNUC__) && defined(__linux__)
+
+// Linux-only, relying on GNU extensions
 #include <execinfo.h>
 #include <unistd.h>
 #include <stdlib.h>
 
-void stack_trace(std::ostringstream& s)
+void get_stack_trace(std::ostringstream& s, const int skip = 0)
 {
     #define BT_BUF_SIZE 1024
     void *buffer[BT_BUF_SIZE];
     int bt_len = backtrace(buffer, BT_BUF_SIZE);
     char **bt = backtrace_symbols(buffer, bt_len);
```

Comment:
This only gives the mangled symbols, the hexadecimal offset and the address. If we also want file name and line number we cannot avoid using other libraries (`elfutils` and `dwarf`), and it takes maybe one hundred more lines of code. I will add it later maybe, but ATM I guess will suffice.